tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,261 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from unittest import mock
+
+import pytest
+import torch
+
+from pytorch_lightning import Callback, seed_everything, Trainer
+from pytorch_lightning.accelerators import HPUAccelerator
+from pytorch_lightning.plugins import HPUPrecisionPlugin
+from pytorch_lightning.strategies.hpu_parallel import HPUParallelStrategy
+from pytorch_lightning.strategies.single_hpu import SingleHPUStrategy
+from pytorch_lightning.utilities import _HPU_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.datamodules import ClassifDataModule
+from tests.helpers.runif import RunIf
+from tests.helpers.simple_models import ClassificationModel
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.core as htcore  # noqa: F401
+
+    os.environ["PL_TORCH_DISTRIBUTED_BACKEND"] = "hccl"
+
+
+@RunIf(hpu=True)
+def test_availability():
+    assert HPUAccelerator.is_available()
+
+
+@pytest.mark.skipif(_HPU_AVAILABLE, reason="test requires non-HPU machine")
+@mock.patch("pytorch_lightning.accelerators.hpu.HPUAccelerator.is_available", return_value=True)
+def test_fail_if_no_hpus():
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(accelerator="hpu", devices=1)
+
+
+@RunIf(hpu=True)
+def test_accelerator_selected():
+    trainer = Trainer(accelerator="hpu")
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+
+
+@RunIf(hpu=True)
+def test_all_stages(tmpdir, hpus):
```

Comment:
Mind adding docstring to the tests?

Reply:
is this test parametrized or how arguments are passed in?

Reply:
Through the script.